### DISCUSSION TITLE
GPU-accelerated environment

### DISCUSSION THREAD

**Discussion by cpwan** [thu-ml/tianshou#764] [Q&A]
Hi there, 
Does tianshou support GPU-accelerated environments (such as [Issac Gym](https://developer.nvidia.com/isaac-gym) )?
By _support_, I mean whether the GPU-accelerated environment can interact with the neural network model all on GPU, without converting the states/ action back and forth between devices.
I have been using other RL libraries but I found that at least 30% of the time was used for moving the data across CPU/GPU. I read from the [docs](https://tianshou.readthedocs.io/en/master/tutorials/cheatsheet.html?highlight=gpu#user-defined-environment-and-different-state-representation) that the `ReplayBuffer` stores the state as `numpy.ndarray` or `dict`. Is it possible for it to store tensors on GPU? It would be great if tianshou can deal with the overhead or if there are some workarounds.

**Comment by Trinkle23897** [comment]
Theoretically it can support. Buffer is implemented on top of Batch, and Batch can naively support torch.Tensor.
